caom2/caom2/obs_reader_writer.py

```diff
@@ -1591,8 +1800,111 @@ def _add_planes_element(self, planes, parent):
             self._add_provenance_element(_plane.provenance, plane_element)
             self._add_metrics_element(_plane.metrics, plane_element)
             self._add_quality_element(_plane.quality, plane_element)
+            
+            if self._output_version >= 22:
+                self._add_position_element(_plane.position, plane_element)
+                self._add_energy_element(_plane.energy, plane_element)
+                self._add_time_element(_plane.time, plane_element)
+                self._add_polarization_element(_plane.polarization, plane_element)
+            
             self._add_artifacts_element(_plane.artifacts, plane_element)
-
+            
+    def _add_position_element(self, position, parent):
+        if position is None:
+            return
+        
+        element = self._get_caom_element("position", parent)
+        self._add_shape_element("bounds", position.bounds, element)
+        if position.dimension:
+            self._add_dimension2d_element("dimension", position.dimension, element)
```

Comment:
I thought I've made a comment here but it must have been lost...

In Python 'if expression' executes if expression is not False, None, 0, '' or empty structure ((), [], {}). But some of these are valid values that you probably want to write. Note that the _add_* methods return if value is None.
Comment applies also applies to lines below.